### DISCUSSION TITLE
Plugin missed in netbox 3.2.7

### DISCUSSION THREAD

**Discussion by vinilex77** [DanSheps/netbox-secretstore#86] [General]
Hello everyone!, i've just installed a fresh netbox from this guide:
https://www.vultr.com/docs/install-and-configure-netbox-on-ubuntu-20-04/ but with latest stable release of repository:
https://docs.netbox.dev/en/stable/installation/3-netbox/
version is 3.2.7 at the moment
Everything seems works fine but netbox_secretstore plugin non working.
from venv is installed:
(venv) root@netbox:/opt/netbox# pip show netbox_secretstore | grep Version
Version: 1.2.0
any ideas?

best regards

**Comment by DanSheps** [comment]
Is the plugin listed in your configuration.py?

**Comment by vinilex77** [comment]
The plugin is installed on local_requirements.txt:
(venv) root@netbox:/opt/netbox# cat local_requirements.txt
napalm
django-storages
netbox-secretstore

**Reply by DanSheps** [reply]
It needs to be listed in your configuration.py as well.

**Reply by vinilex77** [reply]
> It needs to be listed in your configuration.py as well.

Netbox documentation says:
NetBox also supports some optional packages. If desired, these packages must be listed in local_requirements.txt within the NetBox root directory.

could it be that this plugin does not work in Python virtual environments?

**Comment by vinilex77** [comment]
i have try to insert:

PLUGINS = ['netbox-secretstore']

but when server restart:

Aug 05 15:03:42 netbox gunicorn[1609]: django.core.exceptions.ImproperlyConfigured: Unable to import plugin netbox-secretstore: Module not found. Check that the plugin module has been installed within the correct Python environment.

**Reply by DanSheps** [reply]
netbox_secretstore

**Comment by vinilex77** [comment]
in venv it's all ok:

(venv) root@netbox:/opt/netbox/netbox# /opt/netbox/venv/bin/pip install -r /opt/netbox/local_requirements.txt |grep  netbox-secretstore
Requirement already satisfied: netbox-secretstore in /opt/netbox-3.2.7/venv/lib/python3.8/site-packages (from -r /opt/netbox/local_requirements.txt (line 3)) (1.2.0)
Requirement already satisfied: importlib in /opt/netbox-3.2.7/venv/lib/python3.8/site-packages (from netbox-secretstore->-r /opt/netbox/local_requirements.txt (line 3)) (1.0.4)
Requirement already satisfied: pycryptodome in /opt/netbox-3.2.7/venv/lib/python3.8/site-packages (from netbox-secretstore->-r /opt/netbox/local_requirements.txt (line 3)) (3.15.0)

**Reply by DanSheps** [reply]
Please make sure you are following the complete instructions for installation

**Comment by vinilex77** [comment]
Oh my GOD, super noob misunderstanding with underscore.. I punish myself.

Now plugin start ok, but when i try to Add a new user key report this error:

![immagine](https://user-images.githubusercontent.com/110663678/183875315-4bc3c8dd-7641-49bc-ba94-09b4825dcb8e.png)

Secret Roles instead is functionally

**Reply by vinilex77** [reply]
Excuse me, below the logs:
Aug 11 07:37:14 netbox gunicorn[907]: Internal Server Error: /plugins/netbox_secretstore/user-key/edit/
Aug 11 07:37:14 netbox gunicorn[907]: Traceback (most recent call last):
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
Aug 11 07:37:14 netbox gunicorn[907]:     response = get_response(request)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
Aug 11 07:37:14 netbox gunicorn[907]:     response = wrapped_callback(request, *callback_args, **callback_kwargs)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 84, in view
Aug 11 07:37:14 netbox gunicorn[907]:     return self.dispatch(request, *args, **kwargs)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/netbox_secretstore/views.py", line 269, in dispatch
Aug 11 07:37:14 netbox gunicorn[907]:     return super().dispatch(request, *args, **kwargs)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
Aug 11 07:37:14 netbox gunicorn[907]:     return super().dispatch(request, *args, **kwargs)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 119, in dispatch
Aug 11 07:37:14 netbox gunicorn[907]:     return handler(request, *args, **kwargs)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/netbox_secretstore/views.py", line 289, in post
Aug 11 07:37:14 netbox gunicorn[907]:     return render(request, self.template_name, {
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/shortcuts.py", line 24, in render
Aug 11 07:37:14 netbox gunicorn[907]:     content = loader.render_to_string(template_name, context, request, using=using)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/loader.py", line 62, in render_to_string
Aug 11 07:37:14 netbox gunicorn[907]:     return template.render(context, request)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/backends/django.py", line 62, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return self.template.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 175, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return self._render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
Aug 11 07:37:14 netbox gunicorn[907]:     return self.nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]:     return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return compiled_parent._render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
Aug 11 07:37:14 netbox gunicorn[907]:     return self.nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]:     return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return compiled_parent._render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
Aug 11 07:37:14 netbox gunicorn[907]:     return self.nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]:     return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return compiled_parent._render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
Aug 11 07:37:14 netbox gunicorn[907]:     return self.nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]:     return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 63, in render
Aug 11 07:37:14 netbox gunicorn[907]:     result = block.nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]:     return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/defaulttags.py", line 322, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
Aug 11 07:37:14 netbox gunicorn[907]:     return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]:     return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1059, in render
Aug 11 07:37:14 netbox gunicorn[907]:     output = self.filter_expression.resolve(context)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 739, in resolve
Aug 11 07:37:14 netbox gunicorn[907]:     new_obj = func(obj, *arg_vals)
Aug 11 07:37:14 netbox gunicorn[907]:   File "/opt/netbox/netbox/utilities/templatetags/builtins/filters.py", line 79, in meta
Aug 11 07:37:14 netbox gunicorn[907]:     return getattr(model._meta, attr, '')
Aug 11 07:37:14 netbox gunicorn[907]: AttributeError: 'str' object has no attribute '_meta'
Aug 11 07:37:14 netbox gunicorn[907]: Not Found: /favicon.ico
